src/routes/threlte/Arc.svelte

```diff
@@ -1,29 +1,39 @@
-<script>
+<script lang="ts">
 	import { LineGeometry } from 'three/addons/lines/LineGeometry.js'
+	import { LineMaterial } from "three/addons/lines/LineMaterial.js"
 	import { Vector2 } from 'three'
 	import { T } from '@threlte/core'
 	import { flatten, arcToPoints, promoteTo3 } from './projectUtils'
 	import { currentlySelected, currentlyMousedOver, sketchTool } from './stores'
+	import type { EntityType, SketchPoint } from "../../types"
+	import { isEntity } from "../../typeGuards"
 
-	export let id, center, start, end
+	// prettier-ignore
+	const log = (function () { const context = "[Arc.svelte]"; const color="gray"; return Function.prototype.bind.call(console.log, console, `%c${context}`, `font-weight:bold;color:${color};`)})()
 
-	export let dashedLineMaterial,
-		dashedHoveredMaterial,
-		solidLineMaterial,
-		solidHoveredMaterial,
-		solidSelectedMaterial,
-		collisionLineMaterial
+	const type: EntityType = "arc"
+
+	export let id: string, center:SketchPoint, start:SketchPoint, end:SketchPoint
+
+	export let dashedLineMaterial: LineMaterial,
+		dashedHoveredMaterial: LineMaterial,
+		solidLineMaterial: LineMaterial,
+		solidHoveredMaterial: LineMaterial,
+		solidSelectedMaterial: LineMaterial,
+		collisionLineMaterial: LineMaterial
 
-	const type = 'arc'
 
 	let hovered = false
-	$: selected = $currentlySelected.some((e) => e.id === id && e.type === type) ? true : false
+
+	$: selected = $currentlySelected.some((e) => isEntity(e) && e.id === id && e.type === type)
+		? true
+		: false
 
 	const center2 = new Vector2(center.twoD.x, center.twoD.y)
 	const start2 = new Vector2(start.twoD.x, start.twoD.y)
 	const end2 = new Vector2(end.twoD.x, end.twoD.y)
 
-	const points = flatten(promoteTo3(arcToPoints(center2, start2, end2)))
+	const points = flatten(promoteTo3(arcToPoints(center2, start2, end2 /** implicit false */))) // defaulted false in function todo ask Matt
```

Comment:
trying to wrap me head around this; if you have a start and an end and a center the arc can be only one direction?

oh hang on... if it's an arc, rather than a circle, which it is!, then it could be going round the long way or the short way? so how do you know the direction? would a midpoint on the curve of the arc be a solution? it could be a snap point too. although if you do know the direction it's unnecessary data that can be computed when needed :thinking:

Reply:
Yep, given a start and end and center the arc could potentially swing the long way or the short way. 

On the rust side, arcs [already store this info](https://github.com/MattFerraro/CADmium/blob/db497e2dd56c1018ab92f88548c78226e58c5be8/src/rust/cadmium/src/sketch/mod.rs#L1466).

This is important to keep track of because in many cases, a counterclockwise wire indicates that everything inside is part of the shape, and a clockwise wire indicates that everything *outside* is part of the shape. For two concentric circles forming an annulus, I'd expect the outer circle to be made of two counterclockwise arcs and the inner circle to be made of two clockwise arcs.

Reply:
Okay I'll take a look at this shortly. If it involves change that's well outside of typescriptification then I'll leave it for the next pr